### DISCUSSION TITLE
Is there any way to make @vuedx/typescript-plugin-vue work in rollup build?

### DISCUSSION THREAD

**Discussion by jods4** [vuedx/languagetools#228] [Q&A] [answered]
`@vuedx/typescript-plugin-vue` provides a great experience in VS Code, especially if some of your *.vue SFC export more than `default`.

Sadly it seems plugins aren't used when building with `tsc` and so I get many errors at build time, which reduces the usefulness of this plugin a lot.

I'm building with Rollup and `rollup-plugin-typescript2`, is there any way to make the magic work when building as well?

**Comment by znck** [comment]
I haven't explored possibilities in rollup but it should be. I created a CLI tool to serve as replacement for `tsc --noEmit`: https://github.com/znck/vue-developer-experience/blob/main/packages/typecheck/readme.md

**Reply by jods4** [reply]
To provide more context: I'm using Rollup from a Vite build.
I've tweaked the build to use the real Typescript instead of Esbuild (mostly because of const enum support -- and speed is less a concern during build).

I could run your CLI tool first and try to disable typing in `rollup-plugin-typescript2` but that feels a bit like double-duty; and I believe `--transpileOnly` works mostly like Esbuild and handles const enums, type-only imports, etc. a little differently.

Maybe it could work (in webpack as well) if the vuedx magic was packaged as a TS transform? Sorry if it's a stupid idea, I'm not well versed in TS extensibility.

**Comment by znck** [comment, marked as answer]
Tracked as https://github.com/vuedx/languagetools/issues/286
